### DISCUSSION TITLE
Deployment with TYPO3 in docker/ddev

### DISCUSSION THREAD

**Discussion by tzeier** [sourcebroker/deployer-extended-typo3#31] [General]
Has anyone expierience with this deployment scripts in docker/ddev? 
Are there any differences to the installation steps to pay attention to?
I'm developing my TYPO3 projects in docker ddev and have troubles to get this deployer script to work properly.
Can anyone help me?

**Comment by kszymukowicz** [comment]
hi
we doing like that for years now. 
What is your problem exactly?

**Reply by tzeier** [reply]
In the meantime i figured it out. Basically it works now, but it seems to fail with the connection to github. I always get the following message when running the deployer:
`[beta] run export GIT_TERMINAL_PROMPT='0' GIT_SSH_COMMAND='ssh -o StrictHostKeyChecking=accept-new'; [ -f /home/vugiweko/www/tobiaswurmehl.ch/typo3/beta/.dep/repo/HEAD ] || /usr/local/bin/git clone --mirror tobiaswurmehl /home/vugiweko/www/tobiaswurmehl.ch/typo3/beta/.dep/repo 2>&1
[beta] fatal: repository 'tobiaswurmehl' does not exist
[beta]  error  in update_code.php on line 90:
[beta] exit code 128 (Invalid exit argument)
`
Maybe this is, because my github repo is private? Or is there a problem with the ssh key on github?

Is it also possible, to use this deployer script with a private or even a local git repo?

**Comment by kszymukowicz** [comment]
Did you try to clone repo on the remote with the same user as deployer use?

**Reply by tzeier** [reply]
ah nevermind. I tried to set a repository with only the name instead of the github link. Now it seems to work until
`[beta]  RuntimeException  in writable.php on line 13:
[beta]
[beta]   Can't detect http user name.
[beta]   Please setup `http_user` config parameter.`

How can I set the http user name?

**Reply by tzeier** [reply]
nevermind. I found out how to set the http_user. 
Now I get an error with the :bin/php:
`[beta] bash: line 1: /home/vugiweko/www/tobiaswurmehl.ch/typo3/beta/.bin/php: No such file or directory
[beta]  error  in vendors.php on line 31:
[beta] exit code 127 (Command not found)`

Which path should I enter in `->set('bin/php')` in the deploy.php? The path to the php on the server?
